### PR TITLE
PartitionWith: add eagerCancel parameter.

### PR DESCRIPTION
**Pull request by TomasMikula**
When one downstream of `PartitionWith` cancels, events to the other downstream should keep flowing. This is consistent with what `Partition` does.

### CODE REVIEW

**Comment by johanandren** [comment]
In fan out stages we usually call this "eager cancellation" and allow it to be chosen by the user upon creation.

**Comment by TomasMikula** [comment]
I see. I will add `eagerCancel: Boolean` constructor parameter and then update this PR.

**Comment by TomasMikula** [comment]
Updated.

File: contrib/src/main/scala/akka/stream/contrib/PartitionWith.scala [hunk 1]
```diff
@@ -120,6 +133,18 @@ final class PartitionWith[In, Out0, Out1] private (p: In => Either[Out0, Out1])
...
+      override def onDownstreamFinish(): Unit =
+        downstreamFinished()
     })
+
+    private def downstreamFinished(): Unit = {
+      activeDownstreamCount -= 1
+      if (eagerCancel) {
+        completeStage()
+      } else if (activeDownstreamCount == 0) {
+        cancel(in)
```
Comment (johanandren) [inline contrib/src/main/scala/akka/stream/contrib/PartitionWith.scala]: Shouldn't it be `completeStage()` here as well?

Comment (TomasMikula) [reply contrib/src/main/scala/akka/stream/contrib/PartitionWith.scala]: Does it make any difference? By this time, both downstreams have finished, so it remains to only signal cancellation to the upstream.

Comment (johanandren) [reply contrib/src/main/scala/akka/stream/contrib/PartitionWith.scala]: Nah, maybe not.

**Comment by johanandren** [comment]
Fixes #150

**Comment by johanandren** [comment]
Lol, 150 is this PR not an attached issue. NVM.

**Pull request merged**